Source/Utils/WarpXUtil.cpp

```diff
@@ -393,7 +393,7 @@ queryWithParser (const amrex::ParmParse& a_pp, char const * const str, float& va
         // If so, create a parser object and apply it to the value provided by the user.
         std::string str_val;
         Store_parserString(a_pp, str, str_val);
-        val = parseStringtoReal(str_val);
+        val = static_cast<float>(parseStringtoReal(str_val));
```

Comment:
Should the `float&` argument in `queryWithParser` be updated to `amrex::Real&`?
```suggestion
        val = static_cast<amrex::Real>(parseStringtoReal(str_val));
```

Reply:
The argument is intentionally a `float`. There are two versions, one with `float` and one with `double` - this is so that when building WarpX with single precision, the parsing can still be done using double precision.

Reply:
Got ya, thanks!